pyogrio/raw.py

```diff
@@ -175,6 +188,12 @@ def write(
                 f"Could not infer driver from path: {path}; please specify driver explicitly"
             )
 
+    if promote_to_multi is None:
+        promote_to_multi = (
+            geometry_type.startswith("Multi")
+            and driver in DRIVERS_NO_MIXED_SINGLE_MULTI
+        )
```

Comment:
The reason this code is here is because the automatic driver detection is in raw.py. So, do you suggest to move the driver detection also to geopandas.py?

Reply:
Ah, yes, good point. No strong opinion (cc @brendan-ward). If we want to do both (keep driver inference here, but centralize promote_to_multi logic in geopandas.py), we could also put the driver inference in a helper function and call that in both places.

Reply:
I think this should stay in `raw.py`, since it is for writing in general.  I think we should keep the functionality in `geopandas.py` more specific to getting the data frame ready to write.